lib/posthog/feature_flags.rb

```diff
@@ -4,35 +4,70 @@
 require 'posthog/version'
 require 'posthog/logging'
 require 'digest'
+
 class PostHog
+
+  class InconclusiveMatchError < StandardError
+  end
+
   class FeatureFlagsPoller
     include PostHog::Logging
 
     def initialize(polling_interval, personal_api_key, project_api_key, host)
-      @polling_interval = polling_interval || 60 * 5
+      @polling_interval = polling_interval || 30
       @personal_api_key = personal_api_key
       @project_api_key = project_api_key
-      @host = host || 'app.posthog.com'
+      @host = host
       @feature_flags = Concurrent::Array.new
+      @group_type_mapping = Concurrent::Hash.new
       @loaded_flags_successfully_once = Concurrent::AtomicBoolean.new
 
       @task =
         Concurrent::TimerTask.new(
           execution_interval: polling_interval,
-          timeout_interval: 15
         ) { _load_feature_flags }
 
-      # load once before timer
-      load_feature_flags
-      @task.execute
+      # If no personal API key, disable local evaluation & thus polling for definitions
+      if !@personal_api_key.nil?
+        # load once before timer
+        load_feature_flags
+        @task.execute
+      else
+        logger.info "No personal API key provided, disabling local evaluation"
+        @loaded_flags_successfully_once.make_true
+      end
+    end
+
+    def load_feature_flags(force_reload = false)
+      if @loaded_flags_successfully_once.false? || force_reload
+        _load_feature_flags
+      end
     end
 
-    def is_feature_enabled(key, distinct_id, default_result = false, groups = {})
+    def get_feature_variants(distinct_id, groups={}, person_properties={}, group_properties={})
+
+      request_data = {
+        "distinct_id": distinct_id,
+        "groups": groups,
+        "person_properties": person_properties,
+        "group_properties": group_properties,
+      }
+
+      decide_data = _request_feature_flag_evaluation(request_data)
```

Comment:
Suggestion: Let's work with symbols below rather than strings. 

`symbolize_keys` in either here or `_request_feature_flag_evaluation`